lib/yarp/parser_gem_compat.rb

```diff
@@ -0,0 +1,259 @@
+# frozen_string_literal: true
+
+require 'ast'
+
+module YARP
+  module ParserGemCompat
+    class Visitor < ::YARP::BasicVisitor
+      include ::AST::Sexp
+
+      def visit_program_node(node)
+        convert node.statements
+      end
+
+      def visit_statements_node(node)
+        statements = node.body
+        if statements.size == 1
+          convert statements[0]
+        else
+          s(:begin, *convert_array(statements))
+        end
+      end
+
+      private def visit_statements_node_no_single(node)
+        s(:begin, *convert_array(node.body))
+      end
+
+      def visit_begin_node(node)
+        if node.rescue_clause
+          s(:kwbegin, s(:rescue, convert(node.statements), convert(node.rescue_clause)), convert_or_nil(node.else_clause))
+        else
+          super
+        end
+      end
+
+      def visit_rescue_node(node)
+        s(:resbody, s(:array, *convert_array(node.exceptions)), convert_or_nil(node.reference), convert(node.statements))
+      end
+
+      def visit_if_node(node)
+        s(:if, convert(node.predicate), convert_or_nil(node.statements), convert_or_nil(node.consequent))
+      end
+
+      def visit_unless_node(node)
+        s(:if, convert(node.predicate), convert_or_nil(node.consequent), convert_or_nil(node.statements))
+      end
+
+      def visit_else_node(node)
+        convert(node.statements)
+      end
+
+      def visit_next_node(node)
+        s(:next)
+      end
+
+      def visit_module_node(node)
+        s(:module, convert(node.constant_path), convert_or_nil(node.body))
+      end
+
+      def visit_class_node(node)
+        s(:class, convert(node.constant_path), convert_or_nil(node.superclass), convert_or_nil(node.body))
+      end
+
+      def visit_def_node(node)
+        parameters = convert_or_nil(node.parameters) || s(:args)
+        if node.receiver
+          s(:defs, convert(node.receiver), node.name.to_sym, parameters, convert_or_nil(node.body))
+        else
+          s(:def, node.name.to_sym, parameters, convert_or_nil(node.body))
```

Comment:
The parser gem has different types for this. It seems much nicer unified as in YARP as they are quite similar.